typed-racket-lib/typed-racket/private/parse-type.rkt

```diff
@@ -1507,10 +1507,9 @@
   ;; Merge all the non-duplicate entries from the parent types
   (define (merge-clause parent-clause clause)
     (for/fold ([clause clause])
-              ([(k v) (in-dict parent-clause)])
-      (if (dict-has-key? clause k)
```

Comment:
Wow, this is really impressive! I guess it only works if one branch is the iterator identifier?

Reply:
Correct. I added the rule for this one in jackfirth/resyntax#343.